
<launch>
    <arg name="config_path" default="$(find agri_resources)/config/drones/duckey/orin_nx_distrib.yaml" />
    <arg name="depth_scaling_factor" default="0.001" />
    <arg name="inverse_depth" default="false" />
    <arg name="camera_coordinate" default="false" />
    <arg name="identity_threshold" default="0.1" />
    <arg name="tracking_distance" default="0.5" />
    <arg name="lost_time_threshold" default="3.0" />
    <arg name="world_frame_id" default="world" />
    <arg name="self_id" default="0" />

    <node name="estimate_manager" pkg="pose_estimator" type="pose_estimator" output="screen">
        <param name="config_file" type="string" value="$(arg config_path)" />
        <param name="depth_scaling_factor" type="double" value="$(arg depth_scaling_factor)" />
        <param name="inverse_depth" type="bool" value="$(arg inverse_depth)" />
        <param name="camera_coordinate" type="bool" value="$(arg camera_coordinate)" />
        <param name="identity_threshold" type="double" value="$(arg identity_threshold)" />
        <param name="tracking_distance" type="double" value="$(arg tracking_distance)" />
        <param name="lost_time_threshold" type="double" value="$(arg lost_time_threshold)" />
        <param name="world_frame_id" type="string" value="$(arg world_frame_id)" />
        <param name="capture_area_margin" type="double" value="1.5" />
        <remap from="depth" to="/camera/aligned_depth_to_color/image_raw" />
        <remap from="flower_poses" to="pose_estimator/flower_poses"/>
        <remap from="odom" to="/d2vins/odometry" />
        <remap from="target_flower_pose" to="/traj_server/planning/track_pose" />
    </node>

    <node name="flower_detector" pkg="pose_estimator" type="flower_detector.py" output="screen">
        <param name="drone_id" type="int" value="$(arg self_id)" />
        <param name="conf_threshold" type="double" value="0.3" />
        <param name="estimate_pos" type="bool" value="true" />
        <param name="estimate_att" type="bool" value="true" />
        <param name="estimate_whole_att" type="bool" value="true" />
        <param name="use_tensorrt" type="bool" value="true" />
        <param name="use_fp16" type="bool" value="false" />
        <param name="use_fp16_dla" type="bool" value="true" />
        <param name="use_int8" type="bool" value="false" />
        <param name="pub_image" type="bool" value="true" />
        <param name="verbose" type="bool" value="false" />
        <param name="inference_freq" type="double" value="10.0" />
        <remap from="flower_poses" to="pose_estimator/flower_poses"/>
    </node>
</launch>
